packages/account/src/test-utils/setup-test-provider-and-wallets.ts

```diff
@@ -0,0 +1,91 @@
+import type { ChainConfig } from '@fuel-ts/utils';
+import { defaultChainConfig } from '@fuel-ts/utils';
+import { mergeDeepRight } from 'ramda';
+import type { PartialDeep } from 'type-fest';
+
+import type { ProviderOptions } from '../providers';
+import { Provider } from '../providers';
+import type { WalletUnlocked } from '../wallet';
+
+import { AssetId } from './asset-id';
+import type { LaunchNodeOptions } from './launchNode';
+import { launchNode } from './launchNode';
+import type { WalletConfigOptions } from './wallet-config';
+import { WalletConfig } from './wallet-config';
+
+export interface LaunchCustomProviderAndGetWalletsOptions {
+  /** Configures the wallets that should exist in the genesis block of the `fuel-core` node. */
+  walletConfig?: Partial<WalletConfigOptions>;
+  /** Options for configuring the provider. */
+  providerOptions?: Partial<ProviderOptions>;
+  /** Options for configuring the test node. */
+  nodeOptions?: Partial<
+    Omit<LaunchNodeOptions, 'chainConfig'> & {
+      chainConfig: PartialDeep<ChainConfig>;
+    }
+  >;
+}
+
+const defaultWalletConfigOptions: WalletConfigOptions = {
+  count: 2,
+  assets: [AssetId.A, AssetId.B],
+  coinsPerAsset: 1,
+  amountPerCoin: 10_000_000_000,
+  messages: [],
+};
+
+export interface SetupTestProviderAndWalletsReturn extends Disposable {
+  wallets: WalletUnlocked[];
+  provider: Provider;
+  cleanup: () => void;
+}
+
+/**
+ * Launches a test node and creates wallets for testing.
+ * If initialized with the `using` keyword, the node will be killed when it goes out of scope.
+ * If initialized with `const`, manual disposal of the node must be done via the `cleanup` function.
+ *
```

Comment:
Can you add some documentation for both of these cases with code snippets in our docs?

Reply:
I explained the `using` keyword in the very first example and added an **IMPORTANT** mention around `const`/`let` in the same line [here](https://github.com/FuelLabs/fuels-ts/pull/1356/files#diff-af7951c01e94cdb5779ab49a5313e5cf16c9d98b71dd445f8254095eb594cef7R26).